### PR TITLE
Port auxiliary module to pytest

### PR DESCRIPTION
**Pull request by utkbansal**
Fixes #

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by utkbansal** [comment]
@kain88-de @richardjgowers Have a look at this.

**Comment by richardjgowers** [comment]
@utkbansal it looks like you might have to turn `BaseAuxReference` into a fixture.  Then the tests in `BaseAuxReader` should all be parametrised to use it, rather than referring to it through self.

I wouldn't blindly put everything from `BaseAuxReference` into the fixture though, there's a lot of junk in there that could get moved out, for example `iter_list` isn't actually reference data, and that could exist only in the test methods which use it.

**Comment by utkbansal** [comment]
I'm guessing, the error is arising because `setUp` can't take arguments?!

**Comment by utkbansal** [comment]
@richardjgowers Okay, I'll try that.

Interesting to note that the `minimal` build exits once the `pytest` suite fails and it doesn't try to run the `nose` test suite.

File: testsuite/MDAnalysisTests/auxiliary/base.py [hunk 1]
```diff
@@ -34,14 +37,17 @@ def test_get_bad_auxreader_format_raises_ValueError():
     # should raise a ValueError when no AuxReaders with match the specified format
     mda.auxiliary.core.get_auxreader_for(format='bad-format')
 
-class BaseAuxReference(object):
+class BaseAuxReference(TestCase):
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/auxiliary/base.py]: this contains reference data. No need to declare it as a test class

File: testsuite/MDAnalysisTests/auxiliary/test_xvg.py [hunk 2]
```diff
@@ -33,8 +33,11 @@
 from MDAnalysisTests.auxiliary.base import (BaseAuxReaderTest, BaseAuxReference)
 
 class XVGReference(BaseAuxReference):
-    def __init__(self):
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/auxiliary/test_xvg.py]: these are normal classes.

**Comment by utkbansal** [comment]
@kain88-de @richardjgowers I have made some progress. I still have 3 test cases failing. I'm looking into that. A quick review would be helpful. basically what I want to know is - if this the direction we want to proceed in?

**Comment by utkbansal** [comment]
Got those errors, apparently having the scope set to `module` causes some problem. Maybe some methods are updating some attributes?

**Comment by richardjgowers** [comment]
Don't worry about scope yet, that's an optimisation

**Comment by utkbansal** [comment]
This should fix the drop in coverage. I still have to do some cleanup stuff.

**Comment by kain88-de** [comment]
@utkbansal. There is a possible minimal fix in #1440. You can just take the code. No need for commit attribution. Choose better names though! Simply doing the init/setup in a different named method fixes everything. As long as coverage doesn't drop this is OK for now but should be fixed with a better solution long term.

**Comment by utkbansal** [comment]
@kain88-de @richardjgowers All green and no drop in coverage! Have cleaned this up. Please review.

File: testsuite/MDAnalysisTests/auxiliary/test_xvg.py [hunk 3]
```diff
@@ -50,28 +53,36 @@ def __init__(self):
...
 class TestXVGReader(BaseAuxReaderTest):
-    def __init__(self):
-        reference = XVGReference()
-        super(TestXVGReader, self).__init__(reference)
-
-    @raises(ValueError)
-    def test_changing_n_col_raises_ValueError(self): 
+    @staticmethod
+    @pytest.fixture()
+    def ref():
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/auxiliary/test_xvg.py]: what happens when I define a global fixture `ref` in the same file?

File: testsuite/MDAnalysisTests/auxiliary/test_xvg.py [hunk 4]
```diff
@@ -50,28 +53,36 @@ def __init__(self):
...
+        return XVGReference()
+
+    @staticmethod
+    @pytest.fixture()
+    def reader(ref):
+        return ref.reader(ref.testdata, initial_time=ref.initial_time,
+                            dt=ref.dt, auxname=ref.name,
+                            time_selector=None, data_selector=None)
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/auxiliary/test_xvg.py]: indentation

File: testsuite/MDAnalysisTests/auxiliary/test_xvg.py (hunk 3 above)
Comment (utkbansal) [inline testsuite/MDAnalysisTests/auxiliary/test_xvg.py]: I tried making both of them global. 3 tests failed. So I switched back.

**Comment by utkbansal** [comment]
@richardjgowers Should be ready to be merged now.

File: testsuite/MDAnalysisTests/auxiliary/test_xvg.py (hunk 3 above)
Comment (kain88-de) [inline testsuite/MDAnalysisTests/auxiliary/test_xvg.py]: So if I now create a global fixture the tests will break?

**Comment by utkbansal** [comment]
Can someone explain why `test_xvg_bz2`  function has the same name as another function just above it!? https://github.com/utkbansal/mdanalysis/blob/27a9cb2ed3a0ad34b19671c28522c6a4c7c4a365/testsuite/MDAnalysisTests/auxiliary/test_xvg.py#L132

**Comment by kain88-de** [comment]
> Can someone explain why test_xvg_bz2 function has the same name as another function just above it!?

Looks like a mistake. You can rename the functions.

File: testsuite/MDAnalysisTests/auxiliary/test_xvg.py [hunk 5]
```diff
@@ -81,28 +96,44 @@ def __init__(self):
...
+        reader.next()
+        assert_equal(reader.step, 0)
+
 
 def test_xvg_bz2():
     reader = mda.auxiliary.XVG.XVGReader(XVG_BZ2)
     assert_array_equal(reader.read_all_times(), np.array([0., 50., 100.]))
 
+
 def test_xvg_bz2():
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/auxiliary/test_xvg.py]: If you rename  the function so that the test is run as well the PR is good to go.

**Comment by utkbansal** [comment]
@richardjgowers @kain88-de Can be merged now!

**Comment by richardjgowers** [comment]
Another one down!

**Pull request merged**